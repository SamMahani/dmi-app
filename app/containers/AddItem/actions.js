import { ADD_ITEM, ADD_ITEM_SUCCESS, ADD_ITEM_ERROR } from './constants';

export function addItem(item) {
  return {
    type: ADD_ITEM,
    item,
  };
}

export function addItemSuccess(success) {
  return {
    type: ADD_ITEM_SUCCESS,
    success,
  };
}

export function addItemError(error) {
  return {
    type: ADD_ITEM_ERROR,
    error,
  };
}
