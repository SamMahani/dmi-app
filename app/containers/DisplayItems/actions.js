import { GET_ITEMS, GET_ITEMS_SUCCESS, GET_ITEMS_ERROR } from './constants';

export function getItems() {
  console.log('Get items got triggered!');
  return {
    type: GET_ITEMS,
  };
}

export function getItemsSuccess(items) {
  return {
    type: GET_ITEMS_SUCCESS,
    items,
  };
}

export function getItemsError() {
  return {
    type: GET_ITEMS_ERROR,
  };
}
