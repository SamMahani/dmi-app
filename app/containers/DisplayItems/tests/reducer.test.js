import { fromJS } from 'immutable';
import displayItemsReducer from '../reducer';
import { getItemsSuccess } from '../actions';

describe('displayItemsReducer', () => {
  let initialState;
  beforeEach(() => {
    initialState = fromJS({
      items: [],
    });
  });

  it('should return initial state when actions to not match', () => {
    expect(displayItemsReducer(undefined, {})).toMatchSnapshot();
  });

  it('should properly handle getItemsSuccess action', () => {
    const sampleItems = ['item 1', 'item 2', 'item 3'];
    expect(
      displayItemsReducer(initialState, getItemsSuccess(sampleItems)),
    ).toMatchSnapshot();
  });
});
