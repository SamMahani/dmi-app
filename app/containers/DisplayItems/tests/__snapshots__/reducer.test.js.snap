// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displayItemsReducer should properly handle inputQuote action 1`] = `
Immutable.Map {
  "items": Array [
    "item 1",
    "item 2",
    "item 3",
  ],
  "success": true,
}
`;

exports[`displayItemsReducer should return initial state when actions to not match 1`] = `
Immutable.Map {
  "items": Immutable.List [],
  "success": false,
  "error": false,
}
`;
